### PR TITLE
JAMES-3487 Java property: MimeMessageInputStreamSource THRESHOLD

### PR DESCRIPTION
**Pull request by chibenwa**
- Docmentation available
 - Added but commented within jvm.properties file (cf #744)
 
 This is a rework of #285 using #744 to improve discoverability

### CODE REVIEW

**Comment by chibenwa** [comment]
Cc @ottoka @mbaechler @jeantil

**Comment by ottoka** [comment]
Nice to see there is actually a good use for jvm.properties. Did not expect anyone to find one so soon :-)

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/index.adoc [hunk 1]
```diff
@@ -63,3 +63,26 @@ By omitting these files, no extra behaviour is added.
...
+This governs the threshold MimeMessageInputStreamSource relies on for storing MimeMessage content on disk.
+Below, data is stored in memory. Above data is stored on disk.
+Lower values will lead to longer processing time but will minimize heap memory usage. Modern SSD hardware
+should however support a high throughput. Higher values will lead to faster single mail processing at the cost
+of higher heap usage.
+
+
+| james.message.usememorycopy
+|Optional. Boolean. Defaults to false. Recommended value is false.
+Should MimeMessageWrapper use a copy of the message in memory?
```
Comment (Arsnael) [inline server/apps/distributed-app/docs/modules/ROOT/pages/configure/index.adoc:86]:
```suggestion
Should MimeMessageWrapper use a copy of the message in memory? Or should bigger message exceeding james.message.memory.threshold 
be copied to temporary files?
```

File: src/site/xdoc/server/config-system.xml [hunk 2]
```diff
@@ -166,6 +166,23 @@
...
+                    This governs the threshold MimeMessageInputStreamSource relies on for storing MimeMessage content on disk. <br/>
+                    Below, data is stored in memory. Above data is stored on disk.
+                    Lower values will lead to longer processing time but will minimize heap memory usage. Modern SSD hardware
+                    should however support a high throughput. Higher values will lead to faster single mail processing at the cost
+                    of higher heap usage.
+                </dd>
+
+                <dt><strong>james.message.usememorycopy</strong></dt>
+                <dd>Optional. Boolean. Defaults to false. Recommended value is false. <br/>
+                    Should MimeMessageWrapper use a copy of the message in memory?</dd>
```
Comment (Arsnael) [inline src/site/xdoc/server/config-system.xml:183]:
```suggestion
                    Should MimeMessageWrapper use a copy of the message in memory? Or should bigger message exceeding james.message.memory.threshold be copied to temporary files?</dd>
```

**Review by Arsnael — COMMENTED** [review]
I don't know if the last sentence for `james.message.usememorycopy` in the `jvm.properties` files is missing intentionally or not in the doc, so feel free to pick or not those suggestions

**Comment by chibenwa** [comment]
> Did not expect anyone to find one so soon :-)

That's why I was enthousiastic about your proposal ;-)

**Review by Arsnael — APPROVED** [review]

**Comment by jeantil** [comment]
I am glad these settings are more readily available, I can't help but wonder if introducing a parallel (and poorer) configuration file system is really a good move though

**Comment by Arsnael** [comment]
Checkstyle errors:

```
04:42:46,299 [ERROR] src/main/java/org/apache/james/server/core/MimeMessageInputStreamSource.java:[52] (whitespace) EmptyLineSeparator: 'METHOD_DEF' should be separated from previous statement.

04:42:46,299 [ERROR] src/main/java/org/apache/james/server/core/MimeMessageInputStreamSource.java:[58] (whitespace) EmptyLineSeparator: 'VARIABLE_DEF' should be separated from previous statement.
```

**Pull request merged**